atomic_reactor/tasks/common.py

```diff
@@ -122,7 +132,32 @@ def run(self, *args, **kwargs):
             else:
                 signal.signal(signal.SIGTERM, self.throw_task_canceled_exception)
 
-            result = self.execute(*args, **kwargs)
+            reactor_config_path = self._params.config_file or REACTOR_CONFIG_FULL_PATH
+            conf = Configuration(config_path=reactor_config_path)
+
+            userdata = self._params.user_params.get('userdata', None)
+            traceparent = None
+            if userdata and 'traceparent' in userdata:
+                traceparent = userdata.get('traceparent')
+                del userdata['traceparent']
+            if traceparent:
+                os.environ['TRACEPARENT'] = traceparent
```

Comment:
you've tested that this is way how it works to propagate traceparentid?

because I've tested only injection via context